ordersettler/ordersettler.go

```diff
@@ -108,120 +105,39 @@ func (r *OrderSettler) Run(ctx context.Context) {
 	if err != nil {
-		return fmt.Errorf("error getting Cosmos chains: %w", err)
-	}
-	for _, chain := range cosmosChains {
-		if chain.FastTransferContractAddress != "" {
-			chains = append(chains, chain)
-		}
+		return fmt.Errorf("detecting pending settlements: %w", err)
 	}
 
-	for _, chain := range chains {
-		bridgeClient, err := r.clientManager.GetClient(ctx, chain.ChainID)
-		if err != nil {
-			return fmt.Errorf("failed to get client: %w", err)
+	for _, settlement := range pendingSettlements {
+		if r.ordersSeen[settlement.OrderID] {
+			continue
 		}
 
-		fills, err := bridgeClient.OrderFillsByFiller(ctx, chain.FastTransferContractAddress, chain.SolverAddress)
+		sourceChainConfig, err := config.GetConfigReader(ctx).GetChainConfig(settlement.SourceChainID)
 		if err != nil {
-			return fmt.Errorf("getting order fills: %w", err)
+			return fmt.Errorf("getting source chain config: %w", err)
 		}
-		if len(fills) == 0 {
-			// solver has not made any fills on this chain, ignore
-			continue
-		}
-
-		for _, fill := range fills {
-			// continue if order has already been seen
-			if r.ordersSeen[fill.OrderID] {
-				continue
-			}
 
-			sourceChainID, err := config.GetConfigReader(ctx).GetChainIDByHyperlaneDomain(strconv.Itoa(int(fill.SourceDomain)))
-			if err != nil {
-				lmt.Logger(ctx).Warn(
-					"failed to get source chain ID by hyperlane domain. skipping order settlement. it may be unsettled.",
-					zap.Uint32("hyperlaneDomain", fill.SourceDomain),
-					zap.String("orderID", fill.OrderID),
-					zap.Error(err),
-				)
-				r.ordersSeen[fill.OrderID] = true
-				continue
-			}
-			sourceGatewayAddress, err := config.GetConfigReader(ctx).GetGatewayContractAddress(sourceChainID)
-			if err != nil {
-				return fmt.Errorf("getting source gateway address: %w", err)
-			}
-			sourceBridgeClient, err := r.clientManager.GetClient(ctx, sourceChainID)
-			if err != nil {
-				return fmt.Errorf("getting client for chainID %s: %w", sourceChainID, err)
-			}
-
-			height, err := sourceBridgeClient.BlockHeight(ctx)
-			if err != nil {
-				return fmt.Errorf("fetching current block height on chain %s: %w", sourceChainID, err)
-			}
-
-			// ensure order exists on source chain
-			exists, amount, err := sourceBridgeClient.OrderExists(ctx, sourceGatewayAddress, fill.OrderID, big.NewInt(int64(height)))
-			if err != nil {
-				return fmt.Errorf("checking if order %s exists on chainID %s: %w", fill.OrderID, sourceChainID, err)
-			}
-			if !exists {
-				r.ordersSeen[fill.OrderID] = true
-				continue
-			}
-
-			// ensure order is not already filled (an order is only marked as
-			// filled on the source chain once it is settled)
-			status, err := sourceBridgeClient.OrderStatus(ctx, sourceGatewayAddress, fill.OrderID)
-			if err != nil {
-				return fmt.Errorf("getting order %s status on chainID %s: %w", fill.OrderID, sourceChainID, err)
-			}
-			if status != fast_transfer_gateway.OrderStatusUnfilled {
-				r.ordersSeen[fill.OrderID] = true
```

Comment:
the order settlement discovery logic has been refactored into a separate function, but all the logic that is used to skip rpc calls for orders that have already been seen has been discarded. it should not require any general context about the order settler to understand that this function pre and post refactor are not logically equivalent.

Reply:
my bad I accidentally put the ordersSeen check to after the rpc calls, fixed